fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -26,10 +30,40 @@
  * <p>
  * *ALL* instantiated FabricKeyBindings should be registered in
  * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
+ * </p>
+ * <pre><code>
+ * FabricKeyBinding.Builder builder = new FabricKeyBinding.builder();
+ * FabricKeyBinding left = builder
+ *			.id(new identifier("example", "left"))
+ *			.code(Keys.Left)
+ *			.build();
+ * FabricKeyBinding right = builder
+ *			.id(new identifier("example", "right"))
+ *			.code(Keys.Right)
+ *			.build();
+ * KeyBindingRegistry.register(left);
+ * KeyBindingRegistry.register(right);
+ * </code></pre>
  */
 public class FabricKeyBinding extends KeyBinding {
+
+	private final Identifier id;
+
 	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
-		super("key." + id.toString().replace(':', '.'), type, code, category);
+		super(formatKeyName(id), type, code, category);
+		this.id = id;
+	}
+
+	public static String formatKeyName(Identifier id) {
+		return String.format("key.%s.%s", id.getNamespace(), id.getPath());
+	}
+
+	/**
+	 * Original identifier used to register this key.
+	 * May be different from the {@link getId()}!
```

Comment:
Documentation should be formal and descriptive. This should also explain *why* they can differ.
```suggestion
	 * May be different from the {@link getId()}
```

Reply:
Really, having the exclamation mark is more descriptive because it bundles the metadata of the feeling of the sentence.

Reply:
I don't feel like there's really a need to explain why they differ. If you really want to know: It's because Fabric requires you to pass in an identifier, but doesn't actually use it. Instead it converts and mutilates it into a string that gets passed down as the _actual_ identifier.

For a key created with the Identifier `example:mod`, calling `getId()` and parsing that to an Identifier will return back `minecraft:key.example.mod` (hence why I hate that design decision).

Reply:
The root problem is Yarn calling it an ID when it's a translation key lol